bug in admin dashboard which makes the KPI not work

## Description  
ved stats for range: 3months
2026-03-22 12:46:45 [ERROR] endpoints.admin_dashboard_api (admin_dashboard_api:363) - Error fetching questions solved stats: type object 'Submission' has no attribute 'question_instance_id'                         
Traceback (most recent call last):                                                                         
  File "C:\Users\naola\PycharmProjects\thinkly\backend\src\endpoints\admin_dashboard_api.py", line 344, in get_questions_solved_stats                                                                                 
    .join(Submission, QuestionInstance.question_instance_id == Submission.question_instance_id)            
                                                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^             
AttributeError: type object 'Submission' has no attribute 'question_instance_id'. Did you mean: 'user_question_instance_id'?                                                                                          
2026-03-22 12:46:45 [INFO] endpoints.admin_dashboard_api (admin_dashboard_api:384) - Fetching time to solve stats for range: 3months
2026-03-22 12:46:45 [ERROR] endpoints.admin_dashboard_api (admin_dashboard_api:420) - Error fetching time to solve stats: type object 'Submission' has no attribute 'question_instance_id'                            
Traceback (most recent call last):                                                                         
  File "C:\Users\naola\PycharmProjects\thinkly\backend\src\endpoints\admin_dashboard_api.py", line 400, in get_time_to_solve_stats                                                                                    
    .join(Submission, QuestionInstance.question_instance_id == Submission.question_instance_id)            
                                                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^             
AttributeError: type object 'Submission' has no attribute 'question_instance_id'. Did you mean: 'user_question_instance_id'?                                                                                          
>> GET /manage-accounts/users/21/preferences Origin: http://localhost:5173
>> POST /log/client-log Origin: http://localhost:5173
>> POST /log/client-log Origin: http://localhost:5173
2026-03-22 12:46:46 [ERROR] client_logger (log_api:44) - ClientLog url=http://localhost:5173/app/dashboard component=AdminDashboardAPI message=Error fetching questions solved stats: Request failed with status code 500 stack=AxiosError: Request failed with status code 500                                                  
2026-03-22 12:46:46 [INFO] endpoints.manage_accounts_api (manage_accounts_api:220) - Fetched preferences for user ID 21.
2026-03-22 12:46:46 [ERROR] client_logger (log_api:44) - ClientLog url=http://localhost:5173/app/dashboard component=AdminDashboardAPI message=Error fetching time to solve stats: Request failed with status code 500 stack=AxiosError: Request failed with status code 500      

## Steps to Reproduce  
1. Go to '...'
2. Click on '....'
3. Scroll down to '....'
4. See error

## Expected Behavior  
A clear and concise description of what you expected to happen.

## Actual Behavior  
What actually happens instead.

## Screenshots / Logs  
If applicable, add screenshots, logs, or error messages to help explain your problem.

## Additional Context  
Add any other context, links, or references about the problem here.

--- a/backend/src/endpoints/admin_dashboard_api.py
+++ b/backend/src/endpoints/admin_dashboard_api.py
@@ -3,7 +3,7 @@ from sqlalchemy.orm import Session
 from sqlalchemy import func
 from models.schema import (
     UserAccount, Competition, BaseEvent, Question,
-    AlgoTimeSession, QuestionInstance, Submission, UserSession
+    AlgoTimeSession, QuestionInstance, UserQuestionInstance, Submission, UserSession
 )
 from database_operations.database import get_db
 from endpoints.authentification_api import role_required
@@ -341,7 +341,8 @@ async def get_questions_solved_stats(
                 func.count(Submission.submission_id).label('count')
             )
             .join(QuestionInstance, Question.question_id == QuestionInstance.question_id)
-            .join(Submission, QuestionInstance.question_instance_id == Submission.question_instance_id)
+            .join(UserQuestionInstance, QuestionInstance.question_instance_id == UserQuestionInstance.question_instance_id)
+            .join(Submission, UserQuestionInstance.user_question_instance_id == Submission.user_question_instance_id)
             .filter(
                 Submission.status == 'Accepted',
                 Submission.submitted_on >= range_start
@@ -397,7 +398,8 @@ async def get_time_to_solve_stats(
                 ).label('avg_minutes')
             )
             .join(QuestionInstance, Question.question_id == QuestionInstance.question_id)
-            .join(Submission, QuestionInstance.question_instance_id == Submission.question_instance_id)
+            .join(UserQuestionInstance, QuestionInstance.question_instance_id == UserQuestionInstance.question_instance_id)
+            .join(Submission, UserQuestionInstance.user_question_instance_id == Submission.user_question_instance_id)
             .join(BaseEvent, QuestionInstance.event_id == BaseEvent.event_id)
             .filter(
                 Submission.status == 'Accepted',
@@ -537,7 +539,8 @@ async def get_participation_stats(
         def get_submission_count(day_start, day_end):
             query = (
                 db.query(func.count(Submission.submission_id))
-                .join(QuestionInstance, Submission.question_instance_id == QuestionInstance.question_instance_id)
+                .join(UserQuestionInstance, Submission.user_question_instance_id == UserQuestionInstance.user_question_instance_id)
+                .join(QuestionInstance, UserQuestionInstance.question_instance_id == QuestionInstance.question_instance_id)
             )
 
             # Filter by event type

